build-system/app-index/components/main.js

```diff
@@ -0,0 +1,140 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+
+import {Component, h, render} from 'preact';
+import {ExamplesDocumentModeSelect} from './examples-document-mode-select';
+import {ProxyForm} from './proxy-form';
+
+
+class Main extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = JSON.parse(document.querySelector('#initial-state').innerHTML);
+
+    this.state.selectModePrefix = this.state.selectModePrefix || '/';
+  }
+
+  changeDocumentMode = event => {
+    this.setState({
+      selectModePrefix: event.target.value,
+    });
+  }
+
+  renderSelectModeOrNothing() {
+    if (!/^\/examples/.test(this.state.basepath)) {
+      return '';
+    }
+    return (<ExamplesDocumentModeSelect
+      onchange={this.changeDocumentMode}
+      value={this.state.selectModePrefix} />);
+  }
+
+  render() {
+    const documentSelectModeOrNothing = this.renderSelectModeOrNothing();
+
+    return (<div>
+      <wrap>
+        <header>
+          <h1 class="amp-logo">AMP</h1>
+          <div class="right-of-logo">
+            {!this.state.isMainPage && (
+              <a href="/">← Back to main</a>
+            )}
+          </div>
+          <ul>
+            <li>
+              <a href="https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md"
+                target="_blank">
+                Developing
+              </a>
+            </li>
+            <li class="divider">
+              <a href="https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md"
+                target="_blank">
+                Contributing
+              </a>
+            </li>
+            <li>
+              <a href="https://github.com/ampproject/amphtml/"
+                target="_blank">
+                Github
+              </a>
+            </li>
+            <li>
+              <a href="https://travis-ci.org/ampproject/amphtml"
+                target="_blank">
+                Travis
+              </a>
+            </li>
+            <li>
+              <a href="https://percy.io/ampproject/amphtml/"
+                target="_blank">
+                Percy
+              </a>
+            </li>
+          </ul>
+        </header>
+        {this.state.isMainPage && <ProxyForm />}
+      </wrap>
+      <div class="file-list-container">
+        <wrap>
+          <h3 class="code" id="basepath">
+            {this.state.basepath}
+            <a href="https://github.com/ampproject/amphtml/find/master"
+              target="_blank"
+              class="find-icon">
+              Find file
+            </a>
+          </h3>
+          <div class="push-right-after-heading">
+            {documentSelectModeOrNothing}
+            <a href="/~" class="underlined">List root directory</a>
+          </div>
+          <ul class="file-list">
```

Comment:
Making this `file-list` a standalone functional stateless component might be prudent.